src/library/coverart.h

```diff
@@ -30,49 +31,65 @@ struct CoverInfo {
 
     static const quint16 kNullImageHash;
 
-    CoverInfo() : source(UNKNOWN),
-                  type(NONE),
-                  coverLocation(QString()),
-                  trackLocation(QString()),
-                  // This default value is fine: qChecksum(NULL, 0) is 0.
-                  hash(CoverInfo::kNullImageHash) {}
-
-    bool operator==(const CoverInfo& other) const {
-        return other.source == source &&
-                other.type == type &&
-                other.coverLocation == coverLocation &&
-                other.trackLocation == trackLocation &&
-                other.hash == hash;
-    }
-    bool operator!=(const CoverInfo& other) const {
-        return !(*this == other);
+    CoverInfoRelative()
+            : source(UNKNOWN),
+              type(NONE),
+              // This default value is fine: qChecksum(NULL, 0) is 0.
+              hash(kNullImageHash) {
```

Comment:
hash(0) {
   // The default hash value should match the calculated hash for a null image
   DEBUG_ASSERT(CoverArtUtils::calculateHash(QImage()) == hash);
}

The additional constant is not needed. You can simply use CoverInfoRelative().hash for your tests.